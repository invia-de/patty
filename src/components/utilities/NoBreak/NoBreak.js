import React from 'react';
import PropTypes from 'prop-types';
import classnames from '../../../utils/classnames';
import styles from './NoBreak.module.scss';

/**
 * @author [Eric Zieger](mailto:eric.zieger@invia.de)
 */
export default function NoBreak({ ellipsis, className, children, ...other }) {
  return (
    <div
      className={classnames(styles.nobreak, ellipsis && styles.ellipsis)}
      {...other}
    >
      {children}
    </div>
  );
}
NoBreak.propTypes = {
  /** required due to accessibility */
  children: PropTypes.node.isRequired,
  /** cut text and add ... */
  ellipsis: PropTypes.bool
};
