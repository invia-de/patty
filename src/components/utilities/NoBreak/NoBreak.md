```jsx
<NoBreak>
  this text should not break lorem ipsum et dolor sit amet whatever may come
  this text should not break lorem ipsum et dolor sit amet whatever may come
  this text should not break lorem ipsum et dolor sit amet whatever may come
</NoBreak>
<NoBreak ellipsis>
  this text should not break lorem ipsum et dolor sit amet whatever may come
  this text should not break lorem ipsum et dolor sit amet whatever may come
  this text should not break lorem ipsum et dolor sit amet whatever may come
</NoBreak>
```
