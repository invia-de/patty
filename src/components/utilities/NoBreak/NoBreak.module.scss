.nobreak {
  white-space: nowrap;
  max-width: 100%;
  overflow: hidden;
}

.ellipsis {
  text-overflow: ellipsis;
  padding-right: 0.25rem;
}
