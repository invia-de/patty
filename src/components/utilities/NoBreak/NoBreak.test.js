import React from 'react';
import NoBreak from './NoBreak';
import { render } from '@testing-library/react';

test('NoBreak renders', () => {
  const { container } = render(<NoBreak>lorem ipusm</NoBreak>);
  expect(container.firstChild).toMatchSnapshot();
});

test('NoBreak renders with ellipsis styles', () => {
  const { container } = render(<NoBreak>lorem ipusm</NoBreak>);
  expect(container.firstChild).toMatchSnapshot();
});
