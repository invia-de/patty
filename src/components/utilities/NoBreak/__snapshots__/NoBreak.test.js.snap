// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoBreak renders 1`] = `
<div
  class="nobreak"
>
  lorem ipusm
</div>
`;

exports[`NoBreak renders with ellipsis styles 1`] = `
<div
  class="nobreak"
>
  lorem ipusm
</div>
`;
